lib/balena.js

```diff
@@ -0,0 +1,350 @@
+	 * queries to the api for some resource that isn't directly supported
+	 * in the SDK.
+	 *
+	 * @example
+	 * balena.pine.get({
+	 * 	resource: 'release/$count',
+	 * 	options: {
+	 * 		$filter: { belongs_to__application: applicationId }
+	 * 	}
+	 * });
+	 */
+	sdk.pine = pine;
+
+	/**
+	 * @summary Balena errors module
+	 * @member {Object} errors
+	 * @public
+	 * @memberof balena
+	 *
+	 * @description
+	 * The balena-errors module used internally. This is provided primarily for
+	 * convenience, and to avoid the necessity for separate balena-errors
+	 * dependencies. You'll want to use this if you need to match on the specific
+	 * type of error thrown by the SDK.
+	 *
+	 * @example
+	 * balena.models.device.get(123).catch(function (error) {
+	 *   if (error.code === balena.errors.BalenaDeviceNotFound.code) {
+	 *     ...
+	 *   } else if (error.code === balena.errors.BalenaRequestError.code) {
+	 *     ...
+	 *   }
+	 * });
+	 */
+	sdk.errors = errors;
+
+	sdk.version = version;
+
+	return sdk;
+};
+
+/**
+ * @summary Set shared default options
+ * @name setSharedOptions
+ * @public
+ * @function
+ * @memberof balena
+ *
+ * @description
+ * Set options that are used by calls to `balena.fromSharedOptions()`.
+ * The options accepted are the same as those used in the main SDK factory function.
+ * If you use this method, it should be called as soon as possible during app
+ * startup and before any calls to `fromSharedOptions()` are made.
+ *
+ * @params {Object} opts - The shared default options
+ *
+ * @example
+ * balena.setSharedOptions({
+ * 	apiUrl: 'https://api.balena-cloud.com/',
+ * 	builderUrl: 'https://builder.balena-cloud.com/',
+ * 	isBrowser: true,
+ * });
+ */
+getSdk.setSharedOptions = function(opts) {
+	if (globalEnv[BALENA_SDK_HAS_USED_SHARED_OPTIONS]) {
+		console.error(
+			'Shared SDK options have already been used. You may have a race condition in your code.',
+		);
+	}
+
+	if (globalEnv[BALENA_SDK_HAS_SET_SHARED_OPTIONS]) {
+		console.error(
+			'Shared SDK options have already been set. You may have a race condition in your code.',
+		);
+	}
+
+	globalEnv[BALENA_SDK_SHARED_OPTIONS] = opts;
+	globalEnv[BALENA_SDK_HAS_SET_SHARED_OPTIONS] = true;
+};
+
+/**
+ * @summary Create an SDK instance using shared default options
+ * @name fromSharedOptions
+ * @public
+ * @function
+ * @memberof balena
+ *
+ * @description
+ * Create an SDK instance using shared default options set using the `setSharedOptions()` method.
+ * If options have not been set using this method, then this method will use the
+ * same defaults as the main SDK factory function.
+ *
+ * @params {Object} opts - The shared default options
+ *
+ * @example
+ * const sdk = balena.fromSharedOptions();
+ */
+getSdk.fromSharedOptions = function() {
+	const sharedOpts = globalEnv[BALENA_SDK_SHARED_OPTIONS];
+
+	globalEnv[BALENA_SDK_HAS_USED_SHARED_OPTIONS] = true;
+
+	return getSdk(sharedOpts);
+};
+
+modules.exports = getSdk;
```

Comment:
Shouldn't this be
```suggestion
module.exports = getSdk;
```
like it was before?
It seems that the tests are complaining as well.